sql/core/src/test/scala/org/apache/spark/sql/UserDefinedTypeSuite.scala

```diff
@@ -58,6 +61,8 @@ private[sql] class MyDenseVectorUDT extends UserDefinedType[MyDenseVector] {
     datum match {
       case data: Seq[_] =>
         new MyDenseVector(data.asInstanceOf[Seq[Double]].toArray)
+      case data: MyDenseVector =>
+        data
```

Comment:
We do because of this: https://issues.apache.org/jira/browse/SPARK-7186
Without it, a MatchError is raised.